external-crates/move/crates/move-compiler/tests/linter/out_of_bounds_indexing.move

```diff
@@ -0,0 +1,8 @@
+module 0x42::M {
+    use std::vector;
+
+    fun func1() {
+        let arr = vector[1, 2, 3, 4, 5];
+        vector::borrow(&arr, 10);
+    }
+}
```

Comment:
This should have some tests that use index syntax, plus use the vector mutably.

In addition, we need a test for this:

```
    fun func1() {
        let mut arr = vector[1, 2, 3, 4, 5];
        arr.push(6);
        arr.push(7);
        arr.push(8);
        arr.push(9);
        arr.push(10);
        arr.push(11);
        vector::borrow(&arr, 10);
    }

```

Reply:
added more testcase and refactored